src/Psl/Result/Failure.php

```diff
@@ -44,6 +44,20 @@ public function getResult(): void
         throw $this->throwable;
     }
 
+    /**
+     * Unwrap the Result if it is succeeded or return $default value.
+     *
+     * @param D $default
+     *
+     * @return T|D
+     *
+     * @template D
```

Comment:
This function always returns the default.

```suggestion
     * @template D
     *
     * @param D $default
     *
     * @return D
 ```

Reply:
thx for review, good points